src/clj/rems/workflow/dynamic.clj

```diff
@@ -253,7 +286,47 @@
                      :applicantuserid "applicant"
                      :workflow {:type :workflow/dynamic
                                 :handlers ["assistant"]}}
-        requested (apply-command application {:actor "assistant" :decider "deity" :type ::request-decision})
-        decided (apply-command requested {:actor "deity" :decision :approved :type ::decide})]
+        injections {:valid-user? #{"deity"}}
+        requested (apply-command application {:actor "assistant" :decider "deity" :type ::request-decision} injections)
+        decided (apply-command requested {:actor "deity" :decision :approved :type ::decide} injections)]
     (is (= {:decider "deity"} (select-keys requested [:decider :decision])))
     (is (= {:decision :approved} (select-keys decided [:decider :decision])))))
+
+(deftest test-decision-errors
+  (let [application {:state ::submitted
+                     :applicantuserid "applicant"
+                     :workflow {:type :workflow/dynamic
+                                :handlers ["assistant"]}}
+        injections {:valid-user? #{"deity"}}]
+    (is (= {:errors [[:missing-injection :valid-user?]]}
+           (handle-command {:actor "assistant" :decider "deity" :type ::request-decision}
+                           application
+                           {})))
+    (is (= {:errors [[:invalid-user "deity2"]]}
+           (handle-command {:actor "assistant" :decider "deity2" :type ::request-decision}
+                           application
+                           injections)))
+    (is (= {:errors [:unauthorized]}
+           (handle-command {:actor "deity" :decision :approved :type ::decide}
+                           application
+                           injections)))
+    (let [requested (apply-command application {:actor "assistant" :decider "deity" :type ::request-decision} injections)]
+      (is (= {:errors [:unauthorized]}
+             (handle-command {:actor "deity2" :decision :approved :type ::decide}
+                             requested
+                             injections)))
+      (is (= {:errors [[:invalid-decision :foobar]]}
+             (handle-command {:actor "deity" :decision :foobar :type ::decide}
+                             requested
+                             injections))))))
```

Comment:
Add test name: `(testing "other decisions are not possible")`, assuming this test is moved to be right after testing :approved and :rejected.